src/isolate.rs

```diff
@@ -236,29 +268,37 @@ extern "C" fn pre_dispatch(
     )
   };
 
+  let bytes_recv = data_buf.data_len as u64;
+
   let isolate = Isolate::from_void_ptr(user_data);
   let dispatch = isolate.dispatch;
   let (is_sync, op) = dispatch(isolate, control_slice, data_slice);
 
   if is_sync {
     // Execute op synchronously.
     let buf = tokio_util::block_on(op).unwrap();
-    if buf.len() != 0 {
+    let buf_size = buf.len();
+    if buf_size != 0 {
       // Set the synchronous response, the value returned from isolate.send().
       isolate.respond(req_id, buf);
     }
+
+    let state = Arc::clone(&isolate.state);
```

Comment:
Are you sure you need the `Arc::clone` here? From the looks of it you should be able to just do `isolate.state.update_metrics(...)` in the synchronous case.

Reply:
Yes, thanks